File: apps/backend/internal/domain/discover.go
package domain

type DiscoverItem struct {
	CategoryID    string `json:"categoryId"`
	CategoryLabel string `json:"categoryLabel"`
	ID            string `json:"id"`
	Title         string `json:"title"`
	ChannelTitle  string `json:"channelTitle"`
	ThumbnailURL  string `json:"thumbnailUrl"`
	ViewCount     string `json:"viewCount"`
	CountryCode   string `json:"countryCode"`
}

type DiscoverUsecase interface {
	GetDiscoverItems() ([]DiscoverItem, error)
}

File: apps/frontend/types/discover.ts
export interface DiscoverItem {
  categoryId: string;
  categoryLabel: string;
  id: string;
  title: string;
  channelTitle: string;
  thumbnailUrl: string;
  viewCount: string;
  countryCode: string;
}

export interface DiscoverResponse {
  success: boolean;
  data?: DiscoverItem[];
  error?: string;
}
